api/tests/unit/domain/models/TargetProfileWithLearningContent_test.js

```diff
@@ -661,4 +661,90 @@ describe('Unit | Domain | Models | TargetProfileWithLearningContent', () => {
+
+      const stage1 = domainBuilder.buildStage({ id: 'stage1', threshold: 0 });
+      const stage2 = domainBuilder.buildStage({ id: 'stage2', threshold: 40 });
+      const stage3 = domainBuilder.buildStage({ id: 'stage3', threshold: 80 });
+
+      const targetProfile = domainBuilder.buildTargetProfileWithLearningContent({
+        skills: [skill1, skill2, skill3, skill4, skill5, skill6],
+        stages: [stage1, stage2, stage3],
+      });
+
+      // when
+      const skillCountBoundaries = targetProfile.getSkillCountBoundariesFromStages(['stage2']);
+
+      // then
+      expect(skillCountBoundaries).to.deep.equal([
+        { from: 2, to: 3 },
+      ]);
+    });
+
+    it('should return skill count boundaries for the given stage id list', () => {
+      // given
+      const skill1 = domainBuilder.buildTargetedSkill();
+      const skill2 = domainBuilder.buildTargetedSkill();
+      const skill3 = domainBuilder.buildTargetedSkill();
+      const skill4 = domainBuilder.buildTargetedSkill();
+      const skill5 = domainBuilder.buildTargetedSkill();
+      const skill6 = domainBuilder.buildTargetedSkill();
+
+      const stage1 = domainBuilder.buildStage({ id: 'stage1', threshold: 0 });
+      const stage2 = domainBuilder.buildStage({ id: 'stage2', threshold: 40 });
+      const stage3 = domainBuilder.buildStage({ id: 'stage3', threshold: 80 });
+
+      const targetProfile = domainBuilder.buildTargetProfileWithLearningContent({
+        skills: [skill1, skill2, skill3, skill4, skill5, skill6],
+        stages: [stage1, stage2, stage3],
+      });
+
+      // when
+      const skillCountBoundaries = targetProfile.getSkillCountBoundariesFromStages(['stage1', 'stage2']);
+
+      // then
+      expect(skillCountBoundaries).to.deep.equal([
+        { from: 0, to: 1 },
+        { from: 2, to: 3 },
+      ]);
+    });
+
+    it('should return only a from boundary if it’s the last stage', () => {
+      // given
+      const skill1 = domainBuilder.buildTargetedSkill();
+      const skill2 = domainBuilder.buildTargetedSkill();
+      const skill3 = domainBuilder.buildTargetedSkill();
+      const skill4 = domainBuilder.buildTargetedSkill();
+      const skill5 = domainBuilder.buildTargetedSkill();
+      const skill6 = domainBuilder.buildTargetedSkill();
+
+      const stage1 = domainBuilder.buildStage({ id: 'stage1', threshold: 0 });
+      const stage2 = domainBuilder.buildStage({ id: 'stage2', threshold: 40 });
+      const stage3 = domainBuilder.buildStage({ id: 'stage3', threshold: 80 });
+
+      const targetProfile = domainBuilder.buildTargetProfileWithLearningContent({
+        skills: [skill1, skill2, skill3, skill4, skill5, skill6],
+        stages: [stage1, stage2, stage3],
+      });
+
+      // when
+      const skillCountBoundaries = targetProfile.getSkillCountBoundariesFromStages(['stage3']);
+
+      // then
+      expect(skillCountBoundaries).to.deep.equal([
+        { from: 4 },
+      ]);
```

Comment:
*Self-suggestion:* On pourrait également mettre la valeur `to` avec la valeur de `totalSkillsCount` dans le cas du dernier stage. Ça éviterait de faire un cas particulier au niveau du repository et on pourrait toujours utiliser des between. WDYT ?

Reply:
Bonne idée

Reply:
Done